File: 02 백준/01 자료구조/Python/Prob11478.py

"""
title  : 11478. 서로 다른 부분 문자열의 개수 (Silver 3)
time   : 432ms      564ms
memory : 240872KB   317976KB
"""

import sys

input = sys.stdin.readline

def solution():

    N = input().strip()     # 문자열

    answer = set()
    length = len(N)

    for i in range(length):
        for j in range(i+1, length+1):
            answer.add(N[i:j])

    print(len(answer))

solution()

File: 02 백준/01 자료구조/Java/src/Prob11478.java

/**
* title  : 11478. 서로 다른 부분 문자열의 개수 (Silver 3)
* time   : 656ms
* memory : 226260KB
*/

import java.io.*;
import java.util.*;

public class Prob11478 {
    
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String word = br.readLine();
        int length = word.length();
        
        HashSet<String> wordSet = new HashSet<>();

        for (int i = 0; i < length; i++) {
            for (int j = i+1; j <= length; j++) {
                wordSet.add(word.substring(i, j));
            }
        }

        System.out.println(wordSet.size());
    }
}